### PR TITLE
8292182: [TESTLIB] Enhance JAXPPolicyManager to setup required permissions for jtreg version 7 jar

### PR DESCRIPTION
**Pull request by sormuras**
Enhance `JAXPPolicyManager` to setup required permissions for `jtreg` 7 by accept version information in JAR file names.

Find details in https://bugs.openjdk.org/browse/JDK-8292182

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292182](https://bugs.openjdk.org/browse/JDK-8292182): [TESTLIB] Enhance JAXPPolicyManager to setup required permissions for jtreg version 7 jar

### Reviewers
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**) ⚠️ Review applies to [f2046e67](https://git.openjdk.org/jdk/pull/9857/files/f2046e670dd78f3bed483ac0d6a1b57303a7857b)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [f2046e67](https://git.openjdk.org/jdk/pull/9857/files/f2046e670dd78f3bed483ac0d6a1b57303a7857b)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [f2046e67](https://git.openjdk.org/jdk/pull/9857/files/f2046e670dd78f3bed483ac0d6a1b57303a7857b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9857/head:pull/9857` \
`$ git checkout pull/9857`

Update a local copy of the PR: \
`$ git checkout pull/9857` \
`$ git pull https://git.openjdk.org/jdk pull/9857/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9857`

View PR using the GUI difftool: \
`$ git pr show -t 9857`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9857.diff">https://git.openjdk.org/jdk/pull/9857.diff</a>

</details>

### CODE REVIEW

**Review by JoeWang-Java — APPROVED** [review]
Thanks for making it to work with jtreg 7!

**Review by LanceAndersen — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

File: test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 
 import java.net.URL;
+import java.nio.file.Path;
```
Comment (naotoj) [inline test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java:27]: Nit: leftover import?

**Review by naotoj — APPROVED** [review]
LGTM

File: test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java (hunk 1 above)
Comment (sormuras) [reply test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java:27]: Yes. Will remove it.

File: test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java [hunk 2]
```diff
@@ -214,8 +214,9 @@ private boolean isTestMachineryDomain(ProtectionDomain domain) {
         CodeSource cs = (domain == null) ? null : domain.getCodeSource();
         URL loc = (cs == null) ? null : cs.getLocation();
         String path = (loc == null) ? null : loc.getPath();
-        return path != null && TEST_JARS.stream()
-                                .filter(path::endsWith)
+        String name = (path == null) ? null : path.substring(path.lastIndexOf('/') + 1);
```
Comment (jaikiran) [inline test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java:217]: Is there a chance that this path could end with `/` and thus lead to a `IndexOutOfBoundsException` on this call to `substring`?

Comment (sormuras) [reply test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPPolicyManager.java:217]: Yes. Usually the URL of the domain's code source points to a JAR file - but it is possible that a directory is the code source. Then the URL could end with a `/` character.

Will refactor the fix to use `Path::getFileName`.

**Review by LanceAndersen — APPROVED** [review]

**Pull request closed without merging**